### PR TITLE
Refactor & Sonar Pulse (partial)

### LINKED ISSUE #26: Prepare for Captain's Quarters
Make sure we have functions for:
- is a square a part of a sunk ship
- is a square a captain's quarters

### PR DESCRIPTION
**Pull request by astra137**
Resolves #26 and prepares #25 #27 #28 #29 

- Greatly refactors JavaScript
- Delete some Java classes
- Simplify all locations as Square with integers x (column) and y (row)
- Make controller actions polymorphic
- Move functions into more appropriate Java models

### CODE REVIEW

**Comment by astra137** [comment]
I have reviewed commits made by @jorgemanzo resolving #28 and merged them (because they branched off of my branch).

File: src/main/java/assets/Guide.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    if (conclusion) {
+      return 'Reload to start a new game.'
+    } else if (kind) {
+      return `(Press R to rotate)`
+    } else if (sonarMode) {
+      return `(Press S to switch to attack)`
+    } else if (sonarable) {
+      return `(Press S to switch to sonar pulse)`
+    } else {
+      return ''
```
Comment (mi11y) [inline src/main/java/assets/Guide.js:30]: Do you want to leave this like this?

**Review by mi11y — APPROVED** [review]
See comment, otherwise approved

File: src/main/java/assets/Guide.js (hunk 1 above)
Comment (astra137) [reply src/main/java/assets/Guide.js:30]: Surprise sonar gift. I think this is okay.

**Pull request merged**